src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,25 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.Optional;
+import mate.academy.hibernate.relations.dao.MovieDao;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
 
 public class MovieServiceImpl implements MovieService {
+    private final MovieDao dao;
+
+    public MovieServiceImpl(MovieDao dao) {
+        this.dao = dao;
+    }
+
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return dao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        Optional<Movie> result = dao.get(id);
+        return result.orElseGet(Movie::new);
```

Comment:
very  confusing logic... if our entity does not exists, your code would create new. but it's not what we asked for by id. consider using `orElseThrow()`